### PR TITLE
Add documentation and examples for transformers flavor

### PR DESCRIPTION
**Pull request by BenWilson2**
## Related Issues/PRs

 #xxx

## What changes are proposed in this pull request?

Add docs and examples for the transformers flavor.

## How is this patch tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [X] Manual tests (describe details, including test results, below)

Manual validation of docs rendering

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

Added examples for the transformers flavor and documentation guides to the flavor.

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [X] `area/docs`: MLflow documentation pages
- [X] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [X] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by jinzhang21 — APPROVED** [review]
Nice PR, @BenWilson2 !

File: docs/source/models.rst [hunk 1]
```diff
@@ -1939,6 +1939,249 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
...
+================================= ================================= =================
+Conversational                    str or List[str]                  str or List[str]
+Summarization                     str or List[str]                  str or List[str]
+Text Classification               str or List[str]                  str or List[str]
+Text Generation                   str or List[str]                  str or List[str]
+Text2Text Generation              str or List[str]                  str or List[str]
+Token Classification              str or List[str]                  str or List[str]
+Translation                       str or List[str]                  str or List[str]
+ZeroShot Classification*          Dict[str, Union[List[str]|str]]*  str or List[str]
+Table Question Answering**        Dict[str, Union[List[str]|str]]** str or List[str]
```
Comment (harupy) [inline docs/source/models.rst:2000]:
```suggestion
ZeroShot Classification*          Dict[str, List[str] | str]*  str or List[str]
Table Question Answering**        Dict[str, List[str] | str]** str or List[str]
```

Comment (harupy) [reply docs/source/models.rst:2000]: For ZeroShot Classification, is `{"<key1>": "val1", <key1>: ["val2"]}` a valid input? `Dict[str, List[str] | str]` implies it is.

File: examples/transformers/conversational.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+
+conversational_pipeline = transformers.pipeline(model="microsoft/DialoGPT-medium")
+
+signature = mlflow.models.infer_signature(
+    "Hi there, chatbot!",
+    mlflow.transformers.generate_signature_output(conversational_pipeline, "Hi there, chatbot!"),
+)
+
+model_info = mlflow.transformers.log_model(
```
Comment (harupy) [inline examples/transformers/conversational.py:12]:
```suggestion
with mlflow.start_run() as run:
    model_info = mlflow.transformers.log_model(
```

File: examples/transformers/load_components.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import transformers
+import mlflow
+
+translation_pipeline = transformers.pipeline(
+    task="translation_en_to_fr",
+    model=transformers.T5ForConditionalGeneration.from_pretrained("t5-small"),
+    tokenizer=transformers.T5TokenizerFast.from_pretrained("t5-small", model_max_length=100),
+)
+
+model_info = mlflow.transformers.log_model(
```
Comment (harupy) [inline examples/transformers/load_components.py:10]:
```suggestion
with mlflow.start_run() as run:
    model_info = mlflow.transformers.log_model(
```

File: docs/source/models.rst [hunk 4]
```diff
@@ -1939,6 +1939,249 @@ For a ``GroupedPmdarima`` model, an example configuration for the ``pyfunc`` ``p
...
+    # Define the signature
+    signature = mlflow.models.infer_signature(
+        "Hi there, chatbot!",
+        mlflow.transformers.generate_signature_output(
+            conversational_pipeline, "Hi there, chatbot!"
+        ),
+    )
+
+    # Log the pipeline
+    model_info = mlflow.transformers.log_model(
```
Comment (harupy) [inline docs/source/models.rst:2038]:
```suggestion
    with mlflow.start_run():
        model_info = mlflow.transformers.log_model(
```

File: examples/transformers/simple.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+tokenizer = transformers.AutoTokenizer.from_pretrained(architecture)
+
+generation_pipeline = transformers.pipeline(
+    task=task,
+    tokenizer=tokenizer,
+    model=model,
+)
+
+with mlflow.start_run() as run:
+    mlflow.transformers.log_model(
```
Comment (harupy) [inline examples/transformers/simple.py:16]:
```suggestion
    model_info = mlflow.transformers.log_model(
```

File: examples/transformers/simple.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+        transformers_model=generation_pipeline,
+        artifact_path="sentence_builder",
+        input_example="keyboard engineer model data science",
+    )
+
+model_uri = f"runs:/{run.info.run_id}/sentence_builder"
+
+sentence_generator = mlflow.pyfunc.load_model(model_uri)
```
Comment (harupy) [inline examples/transformers/simple.py:24]:
```suggestion
sentence_generator = mlflow.pyfunc.load_model(model_info.model_uri)
```

**Review by harupy — APPROVED** [review]
left some comments, otherwise LGTM!

File: examples/transformers/simple.py (hunk 6 above)
Comment (BenWilson2) [reply examples/transformers/simple.py:24]: much nicer :)

File: examples/transformers/simple.py (hunk 5 above)
Comment (BenWilson2) [reply examples/transformers/simple.py:16]: updated!

File: docs/source/models.rst (hunk 4 above)
Comment (BenWilson2) [reply docs/source/models.rst:2038]: good catch

File: examples/transformers/load_components.py (hunk 3 above)
Comment (BenWilson2) [reply examples/transformers/load_components.py:10]: updated

File: docs/source/models.rst (hunk 1 above)
Comment (BenWilson2) [reply docs/source/models.rst:2000]: I like this better. Easier to read :)

**Pull request merged**